internal/transport/http2_server.go

```diff
@@ -380,6 +380,7 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			s.recvCompress = hf.Value
 		case ":method":
 			httpMethod = hf.Value
+			mdata[":method"] = append(mdata[":method"], hf.Value)
```

Comment:
If we are adding this, I believe we also need to add validation that it _is_ POST.

I thought the plan was to synthesize this to be POST?

Reply:
We do have a validation that it is POST. This was actually my first PR merged: https://github.com/grpc/grpc-go/pull/4241.

Reply:
See RBAC Filter PR for me and Menghan's back and forth on this topic.

Reply:
Whoops.  I had forgotten that.  So you might want to change this in the PR description, then, as it's misleading:

> This PR adds a hardcoded :method POST to Server's metadata

--

> See RBAC Filter PR for me and Menghan's back and forth on this topic.

tl;dr so I don't have to go hunting for the right thread?  This seems pretty straightforward to me.

Reply:
I updated the title.

Reply:
Basically, Eric wants a :method POST in the metadata, because RBAC needs to know this for matching logic. However, the codebase had no previous append to the metadata adding :method POST. The only thing we had was it client side, and reading the value and making sure it was POST, but not in the metadata structure itself. Thus, I decided to put the appending of :method POST in the interceptor, but Menghan mentioned that was hacky and it would be better if I did it in transport, as that is more general solution. Since this broke some tests, I decided to do this in a separate PR.

Reply:
Thanks.  SGTM!